src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -77,7 +77,107 @@ object EdgeSchema {
   def stripEdgeRelation(str: String): String = str.split(sep, 2).last
 }
 
+/**
+ * This class is a wrapper around scala.reflect.api.Type that caches results of calls to the Type so
+ * we don't need to go through reflection for future calls. Caching is done through lazy vals in most
+ * cases but TrieMaps are used in places where results build over time.
+ * @param tpe
+ * @tparam T
+ */
+class CachedType[T](val tpe: Type) {
+  def this()(implicit ttag: TypeTag[T]) = this(typeOf[T])
```

Comment:
What does this mean? I don't understand. Further down you're declaring types as `CachedType[_]`, and I don't understand that either.